libraries/chainbase/include/chainbase/allocators.hpp

```diff
@@ -11,14 +11,15 @@
 #include <boost/interprocess/sync/interprocess_sharable_mutex.hpp>
 #include <boost/interprocess/sync/sharable_lock.hpp>
 #include <boost/interprocess/sync/file_lock.hpp>
+#include <boost/thread/locks.hpp>
```

Comment:
I do not see any changes in chainbase that would require adding this include.

Reply:
Everything that currently includes `allocators.hpp` includes `locks.hpp` first (possibly indirectly).  This PR creates `steem_fwd.hpp`, which is supposed to always be included first.  And `steem_fwd.hpp` includes `allocators.hpp`.

So compilation will fail if you remove this `include`.

Reply:
I hate how lazy we have been in the past with implicit inclusions. This is not the first time a refactor has revealed that a header needs to include code that it doesn't and only works because every source file including it happens to include the file it needs.